plugins/repository-azure/src/main/java/org/elasticsearch/cloud/azure/storage/AzureStorageSettings.java

```diff
@@ -107,35 +153,77 @@ public String toString() {
      * @param settings settings to parse
      * @return A tuple with v1 = primary storage and v2 = secondary storage
      */
+    @Deprecated
     public static Tuple<AzureStorageSettings, Map<String, AzureStorageSettings>> parse(Settings settings) {
-        List<AzureStorageSettings> storageSettings = createStorageSettings(settings);
+        List<AzureStorageSettings> storageSettings = createStorageSettingsDeprecated(settings);
         return Tuple.tuple(getPrimary(storageSettings), getSecondaries(storageSettings));
     }
 
-    private static List<AzureStorageSettings> createStorageSettings(Settings settings) {
+    /**
+     * Parses settings and read all settings available under azure.client.*
+     * @param settings settings to parse
+     * @return All the named configurations
+     */
+    public static Map<String, AzureStorageSettings> load(Settings settings) {
```

Comment:
Can you please rename this method and/or `parse` to make it clear this method is for the new style settings and the other one is for the legacy settings?